framework-cloud/framework-openshift/src/main/java/org/kie/cloud/openshift/log/InstancesLogCollectorRunnable.java

```diff
@@ -0,0 +1,118 @@
+package org.kie.cloud.openshift.log;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.kie.cloud.api.deployment.Instance;
+import org.kie.cloud.common.logs.InstanceLogUtil;
+import org.kie.cloud.openshift.deployment.OpenShiftInstance;
+import org.kie.cloud.openshift.resource.Project;
+import org.kie.cloud.openshift.scenario.OpenShiftScenario;
+import org.kie.cloud.openshift.util.OpenshiftInstanceUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class InstancesLogCollectorRunnable implements Runnable {
+
+    private static final Logger logger = LoggerFactory.getLogger(OpenShiftScenario.class);
+
+    private static final Integer DEFAULT_OBERVABLE_BUFFER_IN_SECONDS = 5;
+
+    private Project project;
+    private String logFolderName;
+
+    private Map<OpenShiftInstance, Future<?>> observedInstances = new ConcurrentHashMap<>();
```

Comment:
I was too lazy to open it in my IDE; cannot we declare an exact type here?

Reply:
This has been changed with the direct use of an executor